File: university-service/src/main/java/ua/com/foxminded/courseproject/repository/ScheduleRepository.java
package ua.com.foxminded.courseproject.repository;

import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;
import ua.com.foxminded.courseproject.entity.Schedule;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

@Repository
public interface ScheduleRepository extends CrudRepository<Schedule, UUID> {
    Optional<Schedule> findById(UUID id);

    List<Schedule> findAll();
}

File: university-service/src/main/java/ua/com/foxminded/courseproject/repository/ScheduleRepository.kt
package ua.com.foxminded.courseproject.repository

import org.springframework.data.repository.reactive.ReactiveSortingRepository
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono
import ua.com.foxminded.courseproject.entity.Schedule
import java.util.*

interface ScheduleRepository : ReactiveSortingRepository<Schedule, UUID> {
    override fun findById(id: UUID): Mono<Schedule>
    override fun findAll(): Flux<Schedule>
}